### ISSUE TITLE
Input lag(nvim freeze for one second because of nvim-cmp)

### ISSUE THREAD

**Issue by geril2207** [ray-x/lsp_signature.nvim#276]
I think it's a problem with nvim-cmp, but maybe you could give some workaround.
I faced with some freezes when inputing with lsp_signature, and make some measuring.

I guess that this helper function with calling cmp introduces cause of my behaviour.
![image](https://github.com/ray-x/lsp_signature.nvim/assets/62308020/8a367c62-220e-44e6-a8a4-5a7b122201e3)

https://github.com/ray-x/lsp_signature.nvim/assets/62308020/a4f7740a-1ca4-41c0-b353-fddbc56a895f

**Comment by ray-x** [comment]
Sometime people may have too many cmp sources and slow down the cmp response time.  Could you just enable LSP source in cmp and check if it still happens?

**Comment by geril2207** [comment]
No, with only one source nvim_lsp, it also freezes. I searched some issues and found that my issue is a duplicate of https://github.com/hrsh7th/nvim-cmp/issues/1613 and possible cause described in https://github.com/hrsh7th/nvim-cmp/issues/1613#issuecomment-1595216446.
